include/otf2xx/chrono/convert.hpp

```diff
@@ -113,9 +118,15 @@ namespace chrono
          */
         otf2::chrono::ticks operator()(time_point t) const
         {
-            assert(ticks_per_second == clock::period::den);
+            if (ticks_per_second == clock::period::den)
+            {
+                return ticks(
+                    otf2::chrono::duration_cast<otf2::chrono::duration>(t.time_since_epoch()).count());
+            }
+
+            const double factor = static_cast<double>(clock::period::den) / ticks_per_second;
             return ticks(
-                otf2::chrono::duration_cast<otf2::chrono::duration>(t.time_since_epoch()).count());
+                std::ceil(otf2::chrono::duration_cast<otf2::chrono::duration>(t.time_since_epoch()).count() / factor));
```

Comment:
Why the `ceil`? I'd expect a `round` or a cast.

Reply:
I added a commentin the other branch, which supersedes this PR: https://github.com/tud-zih-energy/otf2xx/blob/experimental-convert-refactor/include/otf2xx/chrono/convert.hpp#L124

You'll still end up with rounding errors.